### DISCUSSION TITLE
Share how you use UBA

### DISCUSSION THREAD

**Discussion by otseng** [eliranwong/UniqueBible#131] [Show and tell]
How are you using UBA?  Share it here.

**Comment by eliranwong** [comment]
I would like to share a bit on new features, verse number single-click and double-click actions.

I think these new features will supercharge bible studies.  It is ok if you don't agree.

These are actually not new.  We have similar implementations in BibleBento.com, Marvel.bible, and our mobile apps.  They now finally land on our desktop version.

Talk about single-click action first.  Default action is "Smart Indexes".  Give you an example, if you want to study John 3:16.  Click its verse number 16, you will get links of resources specifically relevant to John 3:16.  What does it includes?  I encourage you to find it out.  Smart index is actually one of my favourite features.

Regarding double-click action, it is to open our newly designed Master Control window by default, from where you can access nearly most useful UBA features.

Want customisation?  At the moment, you can configure both actions or even remove them either using Focus menu layout or manually editing file config.py.  We may add these options to other menu layout in the future.  It depends on author preferences of individual menu layout.

Just a final note, when you configure these options, please understand that double-clicking action triggers single-click too.  Therefore, we don't recommend you to open Master Control window with single-click if you have a double-clicking action configured too, as the opened window may stand in the way of the displaying resulted from double-click action.

**Comment by onegladiator4u** [comment]
Hey @otseng 
I'm trying to create a Commentary from Notes files.
Now the Chapter of the commentary opens without a problem. How do I modify it to open correct commentary for a range of verse?
For ex:
Now when I click on verse 18 it opens commentary from top like this
![image](https://user-images.githubusercontent.com/48617390/189533896-3d174603-96ea-4f44-bf33-1427590e667d.png)

But i want it like this 
![image](https://user-images.githubusercontent.com/48617390/189533964-99987c82-17e7-4f57-924c-2df81d6631bd.png)

is it possible to do like commentaries like Matthew Henry's Concise Commentary?

**Reply by onegladiator4u** [reply]
i tried to create by editing sqlite commentary by removing all data and add each portion. But i found that it won't parse bible verses

**Reply by otseng** [reply]
I believe you can do that by adding a tag like below before the line you want to jump to:
```
<vid id="v40.1.18"></vid>
```
What's important is the id contains the book number, chapter number, and verse number.  I have no idea why it's a vid tag. but it probably can be any tag.

**Reply by onegladiator4u** [reply]
Oh ok bro. i'll try that.

**Reply by onegladiator4u** [reply]
it just shows the code as it is bro.
![image](https://user-images.githubusercontent.com/48617390/190078319-5eb76334-1777-4a43-8fe4-837e595983fc.png)

**Reply by onegladiator4u** [reply]
Figured it out Oliver bro,
I just need to copy this to notepad++ and save it as .uba file. Problem occurs when saving through note editor of uba. Also for a range of verses we have to use more tags for verses 1-3 
`<vid id="v40.1.1"></vid><vid id="v40.1.2"></vid><vid id="v40.1.3"></vid>`
